### PR TITLE
Smarty and template support in HTML formatter

### PR DESCRIPTION
**Pull request by nickhagen**
### What does this implement/fix? Explain your changes.

Add smarty and tpl file extension support to the default HTML format engine.
### Does this close any currently open issues?
#940
### Any other comments?

Sorry I have not read into your build process to actually test this, but it looks pretty clean and simple to me.
### Checklist

Check all those that are applicable and complete.
- [ ] Merged with latest `master` branch
- [ ] Added examples for testing to [examples/ directory](examples/)
- [ ] Travis CI passes (Mac support)
- [ ] AppVeyor passes (Windows support)

Add smarty and tpl file extensions to default HTML format engine.

### CODE REVIEW

**Comment by nickhagen** [comment]
I am not entirely sure how your grammar system and matching work, but i have updated the PR to include a new coffee file that seems to mirror the functionality of the twig template handler, and reverted the default html grammar.  Thanks, and let me know if that works for you.

**Comment by Glavin001** [comment]
Good work so far @nickhagen.

I'd recommend that you take a look at https://github.com/Glavin001/atom-beautify/blob/master/docs/add-languages-and-beautifiers.md#how-to-add-a-language

File: src/languages/smarty.coffee [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  ###
+  Supported extensions
+  ###
+  extensions: [
+    'smarty'
+  ]
+
+  options: []
+
```
Comment (Glavin001) [inline src/languages/smarty.coffee]: Do you know which of the existing beautifiers support Smarty?

If so, you should add, for example

```javascript
defaultBeautifier: "Pretty Diff"
```

in which case `Pretty Diff` should be the beautifiers name that supports Smartly.

File: src/languages/smarty.coffee [hunk 2]
```diff
@@ -0,0 +1,24 @@
+module.exports = {
```
Comment (Glavin001) [inline src/languages/smarty.coffee]: Need to change https://github.com/Glavin001/atom-beautify/blob/master/src/languages/index.coffee#L14 to add `smartly` (`smart` ==> `require("smartly.coffee")`).

File: src/languages/smarty.coffee [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+  name: "Smarty"
```
Comment (Glavin001) [inline src/languages/smarty.coffee]: Will need to add a Beautifier to support this language.
For example: https://github.com/Glavin001/atom-beautify/blob/master/src/beautifiers/js-beautify.coffee#L9 and https://github.com/Glavin001/atom-beautify/blob/master/src/beautifiers/prettydiff.coffee#L50

File: src/languages/smarty.coffee [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  ]
+
+  ###
+  Supported extensions
+  ###
+  extensions: [
+    'smarty'
+  ]
+
+  options: []
```
Comment (Glavin001) [inline src/languages/smarty.coffee]: `options` is an object not an array. Try `options: {}` instead.

**Comment by Glavin001** [comment]
Also be sure to add examples that can be tested to https://github.com/Glavin001/atom-beautify/tree/master/examples/nested-jsbeautifyrc

**Comment by Glavin001** [comment]
And be sure to update the documentation so that other users can see Smartly is supported!
See https://github.com/Glavin001/atom-beautify/blob/master/docs/add-languages-and-beautifiers.md#options-documentation for details.

```bash
# Update src/options.json file to have new/changed options
npm run postinstall
# Update documentation to include information about those options
npm run docs
```

**Comment by Glavin001** [comment]
I do not see any changes since the last review so I am temporarily removing myself as assignee and going to review other Pull Requests. Let me know when you're ready for re-review! Thanks!

**Comment by Glavin001** [comment]
Closing since there were no changes made after PR review.

**Pull request closed without merging**